modules/images/webp-uploads/load.php

```diff
@@ -704,3 +704,159 @@ function webp_uploads_update_rest_attachment( WP_REST_Response $response, WP_Pos
+ * the top level `sources` into the metadata, in order to ensure the restore process has a reference to the right
+ * images. When `IMAGE_EDIT_OVERWRITE` is defined and is truthy the function would loop into the sources array
+ * and remove any sources that was created as an edited version, the filenames for those cases are constructed out
+ * of `e-{digits}` where `{digits}` is a number of length 13.
+ *
+ * @param int   $attachment_id The ID of the attachment.
+ * @param array $data The current metadata to be stored in the attachment.
+ * @return array The updated metadata of the attachment.
+ */
+function webp_uploads_restore_image( $attachment_id, $data ) {
+	$backup_sources = get_post_meta( $attachment_id, '_wp_attachment_backup_sources', true );
+
+	if ( ! is_array( $backup_sources ) || ! isset( $backup_sources['full-orig'] ) || ! is_array( $backup_sources['full-orig'] ) ) {
+		return $data;
+	}
+
+	// TODO: Handle the case If `IMAGE_EDIT_OVERWRITE` is defined and is truthy remove any edited images if present before replacing the metadata.
+	// See: https://github.com/WordPress/performance/issues/158.
+
+	$data['sources'] = $backup_sources['full-orig'];
+
+	return $data;
+}
+
+/**
+ * Hook fired right after a metadata has been created or updated, this function would look
+ * specifically only for the key: `_wp_attachment_backup_sizes` which is the one used to
+ * store all the backup sizes. This hook is in charge of cleaning up the additional meta
+ * keys stored in the metadata of the attachment and storing the `sources` property in the
+ * previous full size image due this is not a size we need to move the `sources` from the
+ * metadata back into the full size similar as how it's done on the rest of the sizes.
+ *
+ * @since n.e.x.t
+ *
+ * @param int    $meta_id The ID of the meta value stored in the DB.
+ * @param int    $attachment_id The ID of the post used for this meta, in our case the attachment ID.
+ * @param string $meta_name The name of the metadata to be updated.
+ * @param array  $backup_sizes An array with the metadata value in this case the backup sizes.
+ */
+function webp_updated_postmeta( $meta_id, $attachment_id, $meta_name, $backup_sizes ) {
+	// The backup sources array.
+	if ( '_wp_attachment_backup_sizes' !== $meta_name ) {
+		return;
+	}
+
+	$backup_sources = get_post_meta( $attachment_id, '_wp_attachment_backup_sources', true );
+	$backup_sources = is_array( $backup_sources ) ? $backup_sources : array();
+
+	if ( ! isset( $backup_sources['_sources'] ) ) {
+		return;
+	}
+
+	$target = null;
+	foreach ( array_keys( $backup_sizes ) as $size_name ) {
+		// We are only interested in the `full-` sizes.
+		if ( strpos( $size_name, 'full-' ) === false ) {
+			continue;
+		}
+		// If the target already has the sources attributes find the next one.
+		if ( isset( $backup_sources[ $size_name ] ) ) {
+			continue;
+		}
+
+		$target = $size_name;
+		break;
+	}
```

Comment:
Are you sure this code is correct? It looks like backup sizes and backup sources got somehow mixed up in here. If this is indeed like you intended, can you please explain your thinking process? I don't understand why this function is needed.

I think the above backup and restore functions hooked into `wp_update_attachment_metadata` should be sufficient.

Reply:
Thanks for taking a look, absolutely let me take a look at this approach I agree it would simplify things as well the only downside of that approach tho is the fact that would limit the compatibility of the plugin as if we are using `use` PHP 7 would be required but let me verify that.

Reply:
Just take a look and this one and that's right we do need to bump the support in order to make this change to PHP 7.1 I don't think it makes sense as the benefit is only in the structure of the code while cutting support for installations running previous versions of PHP.


But let me know what do you think about that, thanks.

Reply:
PHP's `use` for functions has been supported for quite a long time. I'm not exactly sure when but for sure that is supported in PHP 5.6, maybe even older versions than that. Where did you see it requires PHP 7.1?

Reply:
Great, Looks like is available since PHP 5.3 I was confusing it with the changelog on PHP docs

- https://www.php.net/functions.anonymous

Reply:
Thanks, just updated the code to match your suggested approach instead.